moveit_kinematics/lma_kinematics_plugin/include/moveit/lma_kinematics_plugin/lma_kinematics_plugin.h

```diff
@@ -122,84 +119,62 @@ class LMAKinematicsPlugin : public kinematics::KinematicsBase
-   * [seed_state(redundancy_limit)-consistency_limit,seed_state(redundancy_limit)+consistency_limit]
+   *  @param consistency_limits
    *  @param solution solution configuration
    *  @return true if check succeeds
    */
-  bool checkConsistency(const KDL::JntArray& seed_state, const std::vector<double>& consistency_limit,
-                        const KDL::JntArray& solution) const;
-
-  int getJointIndex(const std::string& name) const;
+  bool checkConsistency(const Eigen::VectorXd& seed_state, const std::vector<double>& consistency_limits,
+                        const Eigen::VectorXd& solution) const;
 
-  int getKDLSegmentIndex(const std::string& name) const;
+  void getRandomConfiguration(Eigen::VectorXd& jnt_array) const;
 
-  void getRandomConfiguration(KDL::JntArray& jnt_array, bool lock_redundancy) const;
-
-  /** @brief Get a random configuration within joint limits close to the seed state
+  /** @brief Get a random configuration within consistency limits close to the seed state
    *  @param seed_state Seed state
-   *  @param redundancy Index of the redundant joint within the chain
-   *  @param consistency_limit The returned state will contain a value for the redundant joint in the range
-   * [seed_state(redundancy_limit)-consistency_limit,seed_state(redundancy_limit)+consistency_limit]
+   *  @param consistency_limits
    *  @param jnt_array Returned random configuration
    */
-  void getRandomConfiguration(const KDL::JntArray& seed_state, const std::vector<double>& consistency_limits,
-                              KDL::JntArray& jnt_array, bool lock_redundancy) const;
-
-  bool isRedundantJoint(unsigned int index) const;
-
-  bool active_; /** Internal variable that indicates whether solvers are configured and ready */
-
-  moveit_msgs::KinematicSolverInfo ik_chain_info_; /** Stores information for the inverse kinematics solver */
-
-  moveit_msgs::KinematicSolverInfo fk_chain_info_; /** Store information for the forward kinematics solver */
+  void getRandomConfiguration(const Eigen::VectorXd& seed_state, const std::vector<double>& consistency_limits,
+                              Eigen::VectorXd& jnt_array) const;
 
-  KDL::Chain kdl_chain_;
-
-  unsigned int dimension_; /** Dimension of the group */
+  bool initialized_;  ///< Internal variable that indicates whether solver is configured and ready
 
-  KDL::JntArray joint_min_, joint_max_; /** Joint limits */
-
-  mutable random_numbers::RandomNumberGenerator random_number_generator_;
+  unsigned int dimension_;                        ///< Dimension of the group
+  moveit_msgs::KinematicSolverInfo solver_info_;  ///< Stores information for the inverse kinematics solver
 
+  const robot_model::JointModelGroup* joint_model_group_;
   robot_state::RobotStatePtr state_;
+  KDL::Chain kdl_chain_;
+  std::unique_ptr<KDL::ChainFkSolverPos> fk_solver_;
+  Eigen::VectorXd joint_min_, joint_max_;  /// joint limits
```

Comment:
Nit: It bugs me that we can't get this easily from the robot_model. It would be nice if that method existed so that we could remove this duplicate data.

Reply:
I think that's a performance optimization. 
Accessing `robot_model_->getJointModel(index)->getVariableBounds()[0].min_position_` is way slower than cached access to the vector.